### PR TITLE
feat: adds TypeScript definitions

### PR DESCRIPTION
**Pull request by Toxicable**
fixes 195

##### Checklist

- [ ] `npm test`, tests passing
- [ ] `npm run test:snap` (to update the snapshot)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +34,7 @@
   "dependencies": {
     "@bcoe/v8-coverage": "^0.2.3",
     "@istanbuljs/schema": "^0.1.2",
+    "@types/istanbul-lib-report": "^3.0.0",
```
Comment (bcoe) [inline package.json]: How complex is the watermark type, I wonder if we could get away with not pulling in the `@types/istanbul-lib-report` dependency, and just including the description inline.

Comment (Toxicable) [reply package.json]: thats fair, inlined and removed the dep

**Comment by Toxicable** [comment]
Tested downstream here https://github.com/bazelbuild/rules_nodejs/pull/1135/files#diff-cf68df009e956fdc1b73ff16773264f1R80
The CI in that PR will fail (as expected) but that TS file compiles which indicates it's working as expected

**Comment by Toxicable** [comment]
It was tested agains tthat commit I just pushed ^

**Pull request merged**